feat!: update flags for parca-agent 0.10.0

morremeyer opened the pull request:
This PR makes the chart compatible with parca-agent 0.10.0.

BREAKING CHANGE: This breaks compatibility with parca-agent < 0.10.0

kakkoyun commented on charts/parca/README.md:
```diff
@@ -59,20 +65,20 @@ helm repo add parca https://parca-dev.github.io/helm-charts
+| agent.image.tag | string | `"v0.10.0-rc.1"` | Overrides the image tag |
 | agent.logLevel | string | `"info"` | Agent log level |
 | agent.nodeSelector | object | `{}` | node selector for scheduling agent pods |
 | agent.podAnnotations | object | `{}` | Additional annotations for pods |
```
There are also some removed flags such as this. We should clean up as well.

morremeyer replied:
I removed the removed `--socket-path` flag.

kakkoyun replied:
It was not just that one, unfortunately. There are several others. You can find the latest flags in here: https://github.com/parca-dev/parca-agent#configuration

rlex commented:
Should we bump to 0.10.0 release and merge?

kakkoyun reviewed (changes requested):
We need to make sure, all the flags are up-to-date and bump the version.

morremeyer commented:
`--pod-label-selector` is documented in https://github.com/parca-dev/parca-agent#kubernetes-label-selector, but does not exist anymore, it is therefore also removed.

morremeyer commented:
I updated the image tag to `v0.10.0`, too. With this, this PR should be ready for merge.

I squashed all changes into once commit since all previous changes had updates and will need to be re-reviewed anyway.

rlex commented:
probably worth bumping to 1.10.1 alraady

morremeyer commented:
Good idea.

morremeyer commented:
Good idea. Anything else to do here?

rlex commented:
i think it's fine. CI is passing, too. Will merge now.
